catlearn/graph_utils.py

```diff
@@ -536,3 +540,129 @@ def generate_random_graph(
     # return a randomly chosen graph in the factory's memory
     idx = random_generator.randint(0, factory.nb_graphs - 1)
     return factory.graphs[idx]
+
+
+def sample(
+        igraph: DirectedGraph[NodeType],
+        sample_vertices_size: int,
+        ranking: Callable[[DirectedGraph[NodeType]], Mapping[NodeType, float]],
+        rng: random.Random) -> DirectedGraph[NodeType]:
+    """
+    Sample a random subgraph of `igraph` with respect to a probability
+    distribution `ranking` over the vertices.
+
+    Params:
+    - igraph: Input graph
+    - sample_vertices_size: number of vertices to sample
+        (with replacement so actual number might be lower)
+    - ranking: probability distribution over the input graph vertices
+    - rng: Random generator
+
+    Returns:
+    A valid random subgraph
+
+    """
+    ranks = list(ranking(igraph).items())
+    vertices = [v for v, _ in ranks]
+    weights = [w for _, w in ranks]
+    sampled_vertices = set(rng.choices(vertices, weights, k=int(sample_vertices_size)))
+    return igraph.subgraph(sampled_vertices)
+
+
+def pagerank_sample(
+        igraph: DirectedGraph[NodeType],
```

Comment:
same comment again on the argument name. Assume the same comment for all other functions.